app/views/templates/ssh/check_update.erb

```diff
@@ -0,0 +1,33 @@
+<%#
+kind: job_template
+name: Check Update - SSH Default
+model: JobTemplate
+job_category: Packages
+description_format: "Check for package updates"
+provider_type: SSH
+%>
+
+<%
+  supported_families = ['Redhat', 'Debian', 'Suse']
+  render_error(N_('Unsupported or no operating system found for this host.')) unless @host.operatingsystem && supported_families.include?(@host.operatingsystem.family)
+
+  command = case @host.operatingsystem.family
+            when 'Redhat'
+              'yum check-update'
```

Comment:
I've verified on my Fedora and `dnf` behaves the same as `yum` for `check-update`. So no special detection for `dnf` required for now as yum alias is always installed for now. See my comment about return code tho.